### DISCUSSION TITLE
Decimal with sqlite.

### DISCUSSION THREAD

**Discussion by VladasZ** [SeaQL/sea-orm#2203] [Q&A] [answered]
Is there a reason why `Decimal` type is represented as `REAL` while using `sqlite`? I tried to use `sqlite` instead of `Postges` and my tests failed because `REAL` is not precise enough. Isn't there a way to represend `Decimal` to be precise in `sqlite`?

![image](https://github.com/SeaQL/sea-orm/assets/10122409/09b65414-62d8-4b35-854e-6ee19c264a21)

**Comment by tyt2y3** [comment]
There was a time when we tried storing decimals as strings. But it is not without caeveats, because the operations like `SUM` or `ORDER BY` becomes awkard. If you want to store decimal as strings, you can make a custom wrapper type.

**Reply by VladasZ** [reply]
Thanks for the answer. I will look into custom wrappers. 
I get that working with numbers stored as strings may be awkward. But isn't whe whole point of `Decimal` to be precise and the precision is the most important thing here? In my opinion current `Decimal` implementation on `sqlite` is broken because it doesn't do what `Decimal` type is supposed to do. Now it just acts as `float` and it is confusing and may be even dangerous if somebody will use it to count real money.

**Reply by tyt2y3** [reply]
This is not a question I have answer for. Probably a question for the authors of SQLite.

**Comment by VladasZ** [comment, marked as answer]
I guess I'll stick with `Postges` then.
